commands/template_pull_test.go

```diff
@@ -127,6 +126,46 @@ func Test_repositoryUrlRemoteRegExp(t *testing.T) {
 	}
 }
 
+func Test_templatePullPriority(t *testing.T) {
+	templateURLs := []struct {
+		name      string
+		envURL    string
+		cliURL    string
+		resultURL string
+	}{
+		{
+			name:      "Use Default URL when none provided",
+			resultURL: DefaultTemplateRepository,
+		},
+		{
+			name:      "Use Env URL when only env provided",
+			envURL:    "https://github.com/user/project.git",
+			resultURL: "https://github.com/user/project.git",
+		},
+		{
+			name:      "Use Cli URL when only cli provided",
+			cliURL:    "git@github.com:user/project.git",
+			resultURL: "git@github.com:user/project.git",
+		},
+		{
+			name:      "Use Cli URL when both cli and env provided",
+			cliURL:    "git@github.com:user/project.git",
+			envURL:    "https://github.com/user/project.git",
+			resultURL: "git@github.com:user/project.git",
+		},
```

Comment:
Do we need a scenario where all three options are provided?  Could be achieved by passing all three into each scenario.

Reply:
Default is always getting passed